src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -1135,6 +1136,53 @@ private void RunConnectRequestHandlerTask(ConnectParams connectParams)
             }).ContinueWithOnFaulted(null);
         }
 
+        /// <summary>
+        /// Handle new change password requests
+        /// </summary>
+        /// <param name="connectParams"></param>
+        /// <param name="requestContext"></param>
+        /// <returns></returns>
+        protected async Task HandleChangePasswordRequest(
+            ChangePasswordParams changePasswordParams,
+            RequestContext<PasswordChangeResponse> requestContext)
+        {
+            Logger.Write(TraceEventType.Verbose, "HandleChangePasswordRequest");
+            PasswordChangeResponse newResponse = new PasswordChangeResponse();
+            try
+            {
+                RunChangePasswordRequestHandlerTask(changePasswordParams);
+                newResponse.result = true;
+            }
+            catch (Exception ex)
+            {
+                newResponse.result = false;
+                newResponse.ErrorMessage = ex.Message;
+                newResponse.Messages = ex.InnerException != null ? ex.InnerException.Message : ex.Message;
+            }
+            await requestContext.SendResult(newResponse);
+        }
+
+        public void RunChangePasswordRequestHandlerTask(ChangePasswordParams changePasswordParams)
+        {
+            // Empty passwords are not valid.
+            Boolean isPasswordEmpty = string.IsNullOrEmpty(changePasswordParams.NewPassword);

```

Comment:
should use C# primitive types when possible, specifically `bool` here.  Though you could remove this variable and put the `string.IsNullOrEmpty` directly in the if statement.